ob-ipython.el

```diff
@@ -297,6 +297,45 @@ a new kernel will be started."
 
 (add-to-list 'org-src-lang-modes '("ipython" . python))
 
+;; Configure jupyter-<language> blocks for each installed kernel
+(let ((jupyter-executable (executable-find "jupyter")))
+  (when jupyter-executable
+    (let ((jupyter-kernelspec-json
+           (cdr (car (json-read-from-string
+                      (shell-command-to-string
+                       (concat jupyter-executable
+                               " kernelspec list --json")))))))
+      (let ((jupyter-kernelspecs nil))
+        (dolist (x jupyter-kernelspec-json)
+          (add-to-list 'jupyter-kernelspecs
+                       (list (car x) (cdr (assoc 'language (assoc 'spec x))))))
+        (dolist (x jupyter-kernelspecs)
+          (set (intern (format "org-babel-default-header-args:jupyter-%S"
+                               (intern (car (cdr x)))))
+               `((:session . ,(car (cdr x)))
+                 (:kernel . ,(symbol-name (car x)))
```

Comment:
Why not use a variable to store the language? These 'cadr' calls appear many times.